app/models/miq_provision/helper.rb

```diff
@@ -1,6 +1,6 @@
 module MiqProvision::Helper
   def hostname_cleanup(name)
-    hostname_length = (source.platform == 'linux') ? 63 : 15
+    hostname_length = (source.try(:platform) == 'linux') ? 63 : 15
```

Comment:
Does it actually change existing behavior? The only kind of object that could be a source before was one that actually responded to 'platform'.